scripts/notebooks/SampleQC.ipynb

```diff
@@ -0,0 +1,2819 @@
+    "    \n",
+    "    # Print summary values \n",
+    "    print(f'Median: {median_filter_type:.5f}')\n",
+    "    print(f'MAD: {mad_filter_type:.5f}')\n",
+    "    print(f'Minimum: {min(filter_type_data):.5f}')\n",
+    "    print(f'Maximum: {max(filter_type_data):.5f}')\n",
+    "    print()\n",
+    "    \n",
+    "    # Create new figure\n",
+    "    fig = plt.figure(figsize=(8, 6))\n",
+    "    fig.patch.set_facecolor('white')\n",
+    "\n",
+    "    # Plot a histogram\n",
+    "    plt.hist(filter_type_data, bins=num_bins, edgecolor='black')\n",
+    "    if log_scale:\n",
+    "        plt.gca().set_yscale('log')\n",
+    "        plt.gca().yaxis.set_major_formatter(matplotlib.ticker.ScalarFormatter())\n",
+    "        plt.gca().yaxis.get_major_formatter().set_scientific(False)\n",
+    "        plt.gca().yaxis.set_minor_formatter(matplotlib.ticker.NullFormatter())\n",
+    "    plt.gca().set_facecolor('white')\n",
+    "    plt.grid(False)\n",
+    "    plt.gca().yaxis.grid(True, zorder=0, color='lightgrey')\n",
+    "\n",
+    "    # Set the title and axes\n",
+    "    plt.title(f\"{filter_name} - {len(filter_type_data)} Samples\", fontsize=16)\n",
+    "    plt.xlabel(f\"{filter_name}\", fontsize=12)\n",
+    "    plt.ylabel(\"Sample Count (Log Scale)\" if log_scale else \"Sample Count\", fontsize=12)\n",
+    "    \n",
+    "    # Additional plots\n",
+    "    plt.axvline(x=np.median(filter_type_data), color='black', linestyle='--')\n",
+    "    legend_handles = [Line2D([0], [0], alpha=1, color='black', linestyle='--', label='Median')]\n",
+    "    if (cutoffs is not None and line_styles is not None):\n",
+    "        for cutoff, line_style in zip(cutoffs, line_styles):\n",
+    "            plt.axvline(x=median_filter_type + cutoff * mad_filter_type, linestyle=line_style, color='grey', label=f'Median + {cutoff}*MAD')\n",
+    "            plt.axvline(x=median_filter_type - cutoff * mad_filter_type, linestyle=line_style, color='grey', label=f'Median - {cutoff}*MAD')\n",
+    "            legend_handles.append(Line2D([0], [0], alpha=1, linestyle=line_style, color='grey', label=f'Median +- {cutoff}*MAD'))\n",
+    "    plt.legend(handles=legend_handles, framealpha=1, labelspacing=0.5)\n",
+    "\n",
+    "    # Save and close figure\n",
+    "    file_name = f\"filtered_histogram_{len(filter_type_data)}_samples.png\"\n",
+    "    if (cutoffs is not None):\n",
+    "        file_name = f\"analysis_histogram_{len(filter_type_data)}_samples.png\"\n",
+    "    file_path = generate_file_path(TLD_PATH, filter_type, file_name)\n",
+    "    save_figure(WS_BUCKET, file_path)\n",
+    "    plt.close()\n",
+    "    \n",
+    "    return file_path"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "hidden": true
+   },
+   "outputs": [],
+   "source": [
+    "def run_analysis(table, filter_type, num_bins, cutoffs=None, line_styles=None, read_length=None, log_scale=False):\n",
```

Comment:
Makes sense. I guess the only reason I still see value in keeping `num_bins` as is is that I would think it's the main histogram argument that users would see value in modifying, as I can imagine a useful value for this parameter would probably vary across cohort sizes. My thought was that for users who aren't too familiar with `**kwargs`, they may not know how they can modify the bin sizes.

Reply:
I haven't updated this accordingly just yet - will do based on what your thoughts are on `num_bins`. If you think it's fine to just remove `num_bins` all together, happy to do that too. Maybe we can add a comment in the QC section description to show how to use `**kwargs`, and `num_bins` can be the example field we use to illustrate how to pass these in.

Reply:
This would be a nice feature to have since the histograms can be hard to plot when there are extreme outliers. The `bins` parameter also lets you set bin intervals explicitly, which can be especially helpful. I'd suggest replacing `num_bins` with `bins` and providing that explicitly (as a kwargs parameter) to highlight its use while adding that flexibility.

Reply:
Updated accordingly - I've done as you suggested, thanks for clarifying. 

The `bins` parameter is now passed in as a **kwarg for Median Coverage, with a comment highlighting this to users. If no `bins` parameter is provided, then it defaults to 50.